### PR TITLE
Fix#5775 - About page now shows Brave URLs instead of Chrome

### PR DESCRIPTION
**Pull request by rahulsnkr**
Resolves brave/brave-browser#5775.

Currently chromium generates the entire static html page and returns it as a single string. So replacing all instances of "chrome" with "brave" before returning the final string should be sufficient.

image
image

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Comment by rahulsnkr** [comment]
Don't have permissions to request a review, so tagging instead - @rebron

**Comment by simonhong** [comment]
@rahulsnkr I think those new link will not work. Did you check clicking the new url works?

**Comment by rahulsnkr** [comment]
@simonhong yes you're right, it didn't work. The console showed the following error:-

image

However replacing only the text between the `a` tags without replacing the `href` attribute is working.

I could also check if we can add an exception such that the browser loads `brave://` URLs directly, but I'll have to research on where to make that change.

**Comment by simonhong** [comment]
> @simonhong yes you're right, it didn't work. The console showed the following error:-
> 
> image
> 
> However replacing only the text between the `a` tags without changing the `href` attribute is working.
> 
> I could also check if there's a way such that the browser loads `brave://` URLs directly, but I'll have to research on where to make that change.

IMO, replacing text inside the `a` tag seems good fix.
Maybe, using `brave` scheme directly need a lot of effort.

**Comment by rahulsnkr** [comment]
> IMO, replacing text inside the `a` tag seems good fix.
> Maybe, using `brave` scheme directly need a lot of effort.

Updated.

File: chromium_src/chrome/browser/ui/webui/about_ui.cc [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "third_party/re2/src/re2/re2.h"
+#define CHROME_INTERNAL_URLS_TO_BRAVE                                 \
+  const std::string CHROME = "Chrome";                                \
+  const std::string BRAVE = "Brave";                                  \
+  const std::string TAG_CHROME_PROTOCOL = ">chrome:";                 \
+  const std::string TAG_BRAVE_PROTOCOL = ">brave:";                   \
+  const std::string SPACE_CHROME_PROTOCOL = " chrome:";               \
```
Comment (simonhong) [inline chromium_src/chrome/browser/ui/webui/about_ui.cc:12]: nit: we don't use uppercase for local variable name = `space_chrome_protocol`

File: chromium_src/chrome/browser/ui/webui/about_ui.cc [hunk 2]
```diff
@@ -0,0 +1,19 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
```
Comment (simonhong) [inline chromium_src/chrome/browser/ui/webui/about_ui.cc:1]: nit: 2021

File: chromium_src/chrome/browser/ui/webui/about_ui.cc [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "third_party/re2/src/re2/re2.h"
```
Comment (simonhong) [inline chromium_src/chrome/browser/ui/webui/about_ui.cc:6]: nit: give a blank here

Comment (simonhong) [inline chromium_src/chrome/browser/ui/webui/about_ui.cc:6]: Need to add `DEPS` file for this new `re2` module here - `src/brave/chromium_src/chrome/browser/ui/webui/DEPS`
```
include_rules = [
  "+third_party/re2",
]
```

You can check missing gn deps by `npm run gn_check` on your local.

**Review by simonhong — COMMENTED** [review]
This changes looks good to me.
WDYT @brave/patch-reviewers ?

File: chromium_src/chrome/browser/ui/webui/about_ui.cc [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+#define CHROME_INTERNAL_URLS_TO_BRAVE                                 \
+  const std::string CHROME = "Chrome";                                \
```
Comment (simonhong) [inline chromium_src/chrome/browser/ui/webui/about_ui.cc:8]: I think some keywords could be more specific. How about making them more specific like below?
```
const std::string header = "Chrome URLs";
...
const std::string chrome_pages_header = "List of Chrome URLs";
...
const std::string chrome_internal_pages_header = "List of chrome://internals pages";
...
```

**Comment by rahulsnkr** [comment]
@simonhong  Addressed all comments and updated.

**Comment by simonhong** [comment]
Kindly ping to @brave/deps-reviewers @brave/patch-reviewers

**Review by mariospr — APPROVED** [review]
I think adding this one-liner patch in `about_ui.cc` is reasonable considering that this HTML content is hardcoded into the `ChromeURLs()` anonymous function, and that surrounding code doesn't seem to change often (2011, last time).

LGTM, thanks!

**Comment by simonhong** [comment]
Got all green from CI for this PR (https://github.com/brave/brave-core/pull/9060)
Thanks @rahulsnkr !

**Pull request merged**